etna/transforms/timestamp/holiday.py

```diff
@@ -222,9 +301,15 @@ def _transform(self, df: pd.DataFrame) -> pd.DataFrame:
         ValueError:
             if transform isn't fitted
         ValueError:
-            if the frequency is greater than daily and this is a ``binary`` or ``categorical`` mode
+            if the frequency is not weekly, monthly, quarterly or yearly in ``days_count`` mode
+        ValueError:
+            if index timestamp is integer and ``in_column`` isn't set
         ValueError:
-            if the frequency is not weekly, monthly, quarterly or yearly and this is ``days_count`` mode
+            if external timestamp isn't datetime
+        ValueError
+            if in ``days_count`` mode external timestamp doesn't have frequency
+        ValueError
+            if in ``days_count`` mode external timestamp doesn't have the same frequency for all segments
         """
         if self._freq is _DEFAULT_FREQ:
```

Comment:
It means, that `binary` and `categorical` modes will not work, when `external_timestamp` is irregular. I think, we want `HolidayTransform` works in this cases.

Reply:
Why do you think so? I set `self_freq` in all cases during `fit`. Look at both `def _fit` and `def fit`.

Reply:
Okey, now I understand why you define `_DEFAULT_FREQ`